File: leetcode/python/0303_range_sum_query_immutable.py
class NumArray(object):

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.list = nums

    def sumRange(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: int
        """
        sum = 0
        for i in range(left, right + 1):
            sum += self.list[i]
        
        return sum


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: leetcode/java/0303_range_sum_query_immutable.java
class NumArray {
    List<Integer> arrayNums = new ArrayList<Integer>();

    public NumArray(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            arrayNums.add(nums[i]);
        }
    }
    
    public int sumRange(int left, int right) {
        int sum = 0;
        for (int i = left; i <= right; i++) {
            sum += arrayNums.get(i);
        }

        return sum;
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */